x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/TemplateParamValidator.java

```diff
@@ -79,7 +83,17 @@ private static void validateWithSchema(JsonSchema jsonSchema, JsonNode jsonNode)
     }
 
     public void validate(Map<String, Object> templateParams) throws ValidationException {
-        validateWithSchema(this.jsonSchema, OBJECT_MAPPER.valueToTree(templateParams));
+
+        JsonNode secondParam = null;
+        try {
+            SpecialPermission.check();
+            secondParam = AccessController.doPrivileged((PrivilegedExceptionAction<JsonNode>) () -> {
+                return OBJECT_MAPPER.valueToTree(templateParams);
+            });
+        } catch (Exception e) {
```

Comment:
Should this throw a `PrivilegedActionException` rather than a generic `Exception`?

Reply:
Based on other usages of `PrivilegedActionException` in the codebase ([example](https://github.com/elastic/elasticsearch/blob/d2dbef506396a6eeae09f9e028de8c95d5b2bc6a/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java#L35)), this should catch `PrivilegedActionException` and throw the cause.